Userland/Utilities/allocate.cpp

```diff
@@ -1,81 +1,85 @@
 /*
  * Copyright (c) 2018-2020, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2023, Liav A. <liavalb@hotmail.co.il>
  *
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
 #include <AK/Format.h>
 #include <AK/Optional.h>
+#include <LibCore/ArgsParser.h>
 #include <LibCore/ElapsedTimer.h>
 #include <LibMain/Main.h>
 #include <unistd.h>
 
-static void usage()
-{
-    warnln("usage: allocate [number [unit (B/KiB/MiB)]]");
-    exit(1);
-}
-
 enum class Unit {
     Bytes,
     KiB,
     MiB,
+    GiB,
 };
 
 ErrorOr<int> serenity_main(Main::Arguments arguments)
 {
-    int count = 50;
-    auto unit = Unit::MiB;
-
-    if (arguments.argc >= 2) {
-        auto number = arguments.strings[1].to_uint();
-        if (!number.has_value()) {
-            usage();
-        }
-        count = number.value();
-    }
-
-    if (arguments.argc >= 3) {
-        if (arguments.strings[2] == "B")
+    unsigned iterations_count = 10;
+    unsigned allocation_size = 100;
+    auto unit = Unit::Bytes;
+    StringView chosen_unit {};
+
+    Core::ArgsParser args_parser;
+    args_parser.add_option(chosen_unit, "Storage Unit", "unit", 'u', "Storage Unit (Base 2)");
```

Comment:
The words "Storage Unit" are repeated twice in one line:

```
-u Storage Unit (Base 2), --unit Storage Unit (Base 2)  Storage Unit
```

Also, isn't this a size unit?